lib/avro/builder/dsl.rb

```diff
@@ -89,6 +89,25 @@ def type(*)
         @last_object = super
       end
 
+      # Several variants:
+      # define_type(:foo, long)
+      # define_type(:bar, { namespace: 'com.example' }, long)
```

Comment:
I like the idea of dropping the namespace argument. It would make this consistent with how we handle namespacing of other named types right?